plugins/inputs/s7comm/type_conversions.go

```diff
@@ -52,11 +52,25 @@ func determineConversion(dtype string) converterFunc {
 		return func(buf []byte) interface{} {
 			return int32(binary.BigEndian.Uint32(buf))
 		}
+	case "LI":
+		return func(buf []byte) interface{} {
+			if len(buf) != 8 {
+				return nil
+			}
+			return int64(binary.BigEndian.Uint64(buf))
+		}
```

Comment:
How about sticking with the structure of the other cases?
```suggestion
		return func(buf []byte) interface{} {
			return int64(binary.BigEndian.Uint64(buf))
		}
```